eventiq_asb/results.py

```diff
@@ -1,28 +1,48 @@
+from abc import ABC, abstractmethod
 from dataclasses import dataclass
 from typing import Union
 
 from azure.servicebus import ServiceBusReceivedMessage
 
 
 @dataclass
-class BaseResult:
+class BaseResult(ABC):
     message: ServiceBusReceivedMessage
 
+    @abstractmethod
+    def dict(self) -> dict:
+        ...
```

Comment:
`dict[str, Any]`. Also `@abstractmethod` should `raise NotImplementedError`

Reply:
Fixed